gax/retry_policy.h

```diff
@@ -73,37 +100,55 @@ class LimitedErrorCountRetryPolicy : public RetryPolicy {
     return (!status.IsPermanentFailure() && (failure_count_++) < max_failures_);
```

Comment:
I can never remember the rules for operator binding precedence and ordering, so to clarify it for myself and others I wanted to make it explicit that the incremented failure count was being compared to max_failures, instead of the failure count being compared and then incremented.

If you think it's better to be explicit and verbose I'll turn it into

```
if (!status.IsPermanentFailure()) {
  failure_count_++;
  if (failure_count_ < max_failures_) {
    return true;
  }
}

return false;
```

Reply:
I would just write:

```C++
return !status.IsPermanentFailure() && failure_count_++ < max_failures_;
```

Or if you are going for the explicit `if()`s then:

```C++
  if (status.IsPermanentFailure()) {
    return false;
  }
  failure_count_++;
  return failure_count_ < max_failures_;
```

Reply:
Went with the removed parentheses.